srml/contract/src/lib.rs

```diff
@@ -14,41 +14,70 @@
-//! A smart-contract is an account with associated code and storage. When such an account receives a message,
-//! the code associated with that account gets executed.
-//!
-//! The code is allowed to alter the storage entries of the associated account,
-//! create smart-contracts or send messages to existing smart-contracts.
-//!
-//! For any actions invoked by the smart-contracts fee must be paid. The fee is paid in gas.
-//! Gas is bought upfront up to the, specified in transaction, limit. Any unused gas is refunded
-//! after the transaction (regardless of the execution outcome). If all gas is used,
-//! then changes made for the specific call or create are reverted (including balance transfers).
-//!
-//! Failures are typically not cascading. That, for example, means that if contract A calls B and B errors
-//! somehow, then A can decide if it should proceed or error.
-//!
-//! # Interaction with the system
-//!
-//! ## Finalization
-//!
-//! This module requires performing some finalization steps at the end of the block. If not performed
-//! the module will have incorrect behavior.
-//!
-//! Thus [`Module::on_finalise`] must be called at the end of the block. The order in relation to
-//! the other module doesn't matter.
-//!
-//! ## Account killing
-//!
-//! When `staking` module determines that account is dead (e.g. account's balance fell below
-//! exsistential deposit) then it reaps the account. That will lead to deletion of the associated
-//! code and storage of the account.
-//!
-//! [`Module::on_finalise`]: struct.Module.html#impl-OnFinalise
+//! ### Public functions ([`Module`])
+//! 
+//! * `contract_fee` - Gets the fee required to create a contract
+//! 
+//! * `call_base_fee` - Gets the fee required for a call into a contract.
+//! 
+//! * `create_base_fee` - Gets the fee required for a create of a contract. 
+//! 
+//! * `gas_price` - Gets the price of one unit of gas.
+//! 
+//! * `max_depth` - Gets the max nesting level of a call/create stack.
+//! 
+//! * `block_gas_limit` - Gets the max gas amount that could be expended per block.
+//! 
+//! * `gas_spent` - Gets the gas spent so far in this block
+//! 
+//! * `current_schedule` - Gets the current cost schedule for contracts.
+//! 
+//! ## Usage
+//! 
+//! The contract module is a work in progress. The following examples show how this contract module can be used to create and call contracts.
+//! 
+//! * [`pDSL`](https://github.com/Robbepop/pdsl) is a domain specific language which enables writing WebAssembly based smart contracts in the Rust programming language. This is a work in progress.
+//! * [`Sample Contract`](https://github.com/pepyakin/substrate-contracts-adder) is a raw contract example that can be instrumented with [`Polkadot UI`](https://polkadot.js.org/apps/). This is a work in progress.
+//! 
+//! ## Related Modules
+//! * [`Balances`](https://crates.parity.io/srml_balances/index.html)
+//! * [`Staking`](https://crates.parity.io/srml_staking/index.html)
```

Comment:
contract module and staking module are unrelated at all

Reply:
This probably comes from the fact that the contract module was a part of staking module back in the day : )

Now, indeed, it is not longer true.